### ISSUE TITLE
Manual install in FF57

### ISSUE THREAD

**Issue by stonecrusher** [ChoGGi/CuteButtons#5]
Hey ChoGGi,
I really tried, but couldn't make it work in FF 57.0 (x64, stable release from today) in Windows 8.1. I followed your instructions exactly, but nothing happens. Of course restarted, put imports before namespace, cared about case sensitivity and also tried a fresh profile and putting `/*AGENT SHEET*/` in the first line but it won't work.

Does it work for you? Where might be the problem?

If I can make it work later, I'll try to change the paths so that all those css files don't clutter my /chrome folder, because I also want to use https://github.com/Aris-t2/CustomCSSforFx
But what I tried was also on a completely clean install.

Thanks for help, Stonecrusher

**Comment by ChoGGi** [comment]
You may have set it up incorrectly, see if this helps (UserContent.css should look the same other than not adding that @namespace line at the bottom).

![firefoxuserchomeexample](https://user-images.githubusercontent.com/1261282/32801413-349641ac-c93a-11e7-944a-5f7ee99f465d.png)


If you want the files in a folder (called say CuteButtonsSVG), then change the paths to:
@import url("CuteButtonsSVG/BlurDisabledIcons.css");
and place the CSS/PNG files in it.

**Comment by stonecrusher** [comment]
Can't find an error.
Fresh install, only addon uBlock is deactivated.

My [userChrome.css](https://pastebin.com/ewTPsN4D)
My [userContent.css](https://pastebin.com/3JBAAiQ1)

[Screenshots](https://imgur.com/a/RzDSo)

No Icons in FF menu, no icons in context menu.
The Icons provided by CuteButtons are quite important to find the desired button fast by just by scanning with the eyes. Otherwise you need to read everything or need to exactly know where it is :-(

So if it works for you I'll try another time in a win7 VM.

**Comment by B00ze64** [comment]
@ChoGGi Hey, you sure we need to import the CSS /before/ the namespace line? I've just never seen anything else saying we needed to add CSS rules before that line (Still with ESR52 and my "inverted" Normal/Bright icons lol, so I haven't tried importing your CSS yet).

**Comment by stonecrusher** [comment]
@B00ze64 
See [here](https://developer.mozilla.org/en-US/docs/Web/CSS/%40import) and [here](https://www.w3.org/TR/REC-CSS2/syndata.html#at-rules).

**Comment by ChoGGi** [comment]
@B00ze64 It's either add them before the @namespace line, or edit each CSS file and remove the @namespace from it :)
Normally it doesn't really matter where you have @import

@stonecrusher It definitely works for me. No idea what's wrong on your end (your OS of choice shouldn't make any difference).
![firefoxicons](https://user-images.githubusercontent.com/1261282/32816153-9da91036-c974-11e7-8ae4-a226b22a38f4.png)

**Comment by ChoGGi** [comment]
@B00ze64 It's either add them before the @namespace line, or edit each CSS file and remove the @namespace from it :)
Normally it doesn't really matter where you have @import

@stonecrusher Ahh you have both of the NoIcons styles :)
Also I am pretty sure you need mosiac.Hover.png to be mosaic.hover.png (case-sens)

**Comment by stonecrusher** [comment]
Ok now I'm happy (yay, icons are back!) but also feel very stupid -.-
To my defense, I didn't even know what those noIcon styles should do and of course overlooked them in the aftermath. Didn't make sense to me to install an iconset to hide icons. Thought it was some kind of fix/compability issue.

I saw you added explaining comments, which is great but furthermore I suggest providing a copy-and-paste-ready version in the readme (fitting for FF57). Some folks don't even know how to comment out.
Thanks a lot!

**Comment by ChoGGi** [comment]
Well I've got all the button/menuitem selectors in one file, and some people only like them on one or the other so :)

They don't need to comment out, just don't include the line/file no?

**Comment by stonecrusher** [comment]
In a way you are right, but I think we can do better.

The main purpose of the addon is / was to add icons to preferably all buttons and menu items. And I guess that's what most people installed and used it for. So when installing manually it should be as easy as possible and already come with a default setting that displays icons (which it doesn't atm).

I imagine something like [this](https://github.com/ChoGGi/CuteButtons/files/1476961/CuteButtons_Manual_Install.zip) and the simple instruction to unzip that directly to your [profile](https://support.mozilla.org/en-US/kb/profiles-where-firefox-stores-user-data) folder.
That should be already enough for most users.

Further rules:
- If  `userChrome.css` and `userContent.css` are already present (usually you need to create them first), do not overwrite them but add `@import url("CuteButtonsOptions.css");` to the very beginning of the already existing files.
- For switching CuteButtons options (one per line), open `CuteButtonsOptions.css` and remove leading `/*` to activate or add leading `/*` to disable that option.
- Variations of the Icons can be found in the CuteButtons directory and can be used to replace `mosaic.png` and `mosaic.hover.png`.

Btw I don't know which *fixing*.css files are still necessary / up to date. Enabling them by default or deleting might make sense. The extra download for manual install also makes sense to prevent confusion about surplus files. If it was a very big project or had weekly updates, a new repo might make sense for maintenance but in this case just overwriting files from the `content` directory to the `CuteButtons` directory and final zipping should be reasonable.

tl;dr (not really)
----
Outcomment

    @import url("NoIconsButtons.css"); /*removes all icons from buttons*/
    @import url("NoIconsMenus.css"); /*removes all icons from menus*/

in `readme.md` to enable icons by default.

**Comment by ChoGGi** [comment]
_In a way you are right, but I think we can do better._

I just added the guide for people that really wanted the icons, so I don't have any plans to get any fancier then it already is.

I do hope that Mozilla adds some sort of API to style the browser, if not well I've got Palemoon so I'm good (for now).

**Comment by stonecrusher** [comment]
So you won't make showing icons a default setting for an addon described as "Adds icons to buttons and/or menus."?

I would like

    /*@import url("NoIconsButtons.css");*/ /*removes all icons from buttons*/
    /*@import url("NoIconsMenus.css");*/ /*removes all icons from menus*/

much better in the instructions for manual install.

> I do hope that Mozilla adds some sort of API to style the browser

Yes and that's not the only API we're still waiting for :-/

Sticking to old versions or small forks never went well for me so far. Bugs piling up and the following change to another program / new version (sometimes years after) will be even harder. But you're right, you'll be fine for now. Maybe in 3 years Chrome is the only browser left ;-)

**Comment by ChoGGi** [comment]
I hope anyone taking the time to manually install it will notice the comments and not copy that file(s) over.

_Sticking to old versions or small forks never went well for me so far._
Well, I also use Chromium, Opera (Presto), Seamonkey; sticking with one ain't no fun :)



Though in three years, it'll probably be just Chrome and Chrome...

_An internet blogs about a new version of a web browser that might be released eventually. The traditional markers of progress are still present: removing popular functionality, breaking the extensions that reimplement it, adding shitware and disabling its removal, and adding more databases. However, in an effort to modernize the project management, much progress has been made in the most important task: being Chrome. Hackernews is dutifully impressed by all of the wonderful progress Firefox has made, but won't use it because it's not Chrome._

**Comment by stonecrusher** [comment]
> I hope anyone taking the time to manually install it will notice the comments and not copy that file(s) over.

I hope that everyone trying to set up this IKEA wardrobe will read the manual first... ;-)

> sticking with one ain't no fun :)

Same here, but switching your main browser nevertheless always hurts :-/

Though I'm not as negative about the new FF as your quote looks like.
I believe them that a major rewrite was necessary. Of course there's still a lack of API and some things that will never work again, but requirements to security and flexibility grew bigger and bigger so they couldn't really maintain that monstrous dinosaur-aged code anymore. I don't really see "shitware" coming with it. I'm happy that specifications will be implemented early and extensions will now last through a lot of version changes because they use the API and not the everchanging internal stuff.

**Comment by ChoGGi** [comment]
Why yes I do read IKEA manuals.

Oh, that's an n-gate quote (probably should've added a link)
http://n-gate.com/hackernews/2017/09/30/0/
I'm more in the middle; if they add more customisation then I like it, but as for now it feels rushed?

**Comment by stonecrusher** [comment]
> yes I do read IKEA manuals.

Sooner or later all people need to do so when it doesn't work... but the goal is to make it so easy that you don't even need a manual, right? Adding softclose-drawers to the wardrobe is optional, but building the thing itself should be selfexplanatory.

Ok, it's not a 1:1 metaphor.

Ah ok I was a bit confused about the `hackernews` in the quote as I usually think of news.ycombinator.com then.

Yes, it feels rushed. On the other hand they decided *two years ago* that they had to change something radically to keep up with chrome browser market shares. That's a lot of time in the IT world. We knew it would come, but I hoped it would be more "complete" or that they will delay the deadline half a year.

I personally tend to use old stuff as long as it works, which is good for the environment and my budget. So sometimes I need a little push or a really cool innovation to buy something new. As the new Firefox doesn't cost me a penny and doesn't produce waste, it's a different thing. I'm openminded then, force-tried it and have a more progressive attitude. The longer you're accustomed to a habit, the rustier you become (no pun intended) . So I am not afraid to get out of my comfort zone and start learning again.

Well, that became a bit generic and off-topic now... sorry ^^

**Comment by ChoGGi** [comment]
_but the goal is to make it so easy that you don't even need a manual, right?_
Nahh, but I just had another person do the same mistake you did, so I made it a bit clearer in the instructions ;)

**Comment by ChoGGi** [comment]
_but the goal is to make it so easy that you don't even need a manual, right?_

Nahh where's the fun in that?

but I just had another person do the same mistake you did, so I made it a bit clearer in the instructions ;)
